health-connect/HealthConnectSample/app/src/main/java/com/example/healthconnectsample/presentation/screen/recordlist/RecordListScreen.kt

```diff
@@ -0,0 +1,169 @@
+import androidx.health.connect.client.records.HeartRateRecord
+import androidx.health.connect.client.records.Record
+import androidx.health.connect.client.records.SpeedRecord
+import androidx.health.connect.client.records.StepsRecord
+import androidx.health.connect.client.records.TotalCaloriesBurnedRecord
+import androidx.health.connect.client.records.metadata.Metadata
+import androidx.health.connect.client.units.Velocity
+import com.example.healthconnectsample.R
+import com.example.healthconnectsample.formatDisplayTimeStartEnd
+import com.example.healthconnectsample.presentation.theme.HealthConnectTheme
+import java.time.Instant
+import java.time.ZonedDateTime
+import java.time.ZonedDateTime.now
+import java.time.temporal.ChronoUnit
+import java.util.UUID
+import kotlin.random.Random
+
+@Composable
+fun RecordListScreen(
+    uid: String,
+    permissions: Set<String>,
+    permissionsGranted: Boolean,
+    recordType: RecordType,
+    seriesRecordsType: SeriesRecordsType,
+    recordList: List<Record>,
+    uiState: RecordListScreenViewModel.UiState,
+    onError: (Throwable?) -> Unit = {},
+    onPermissionsResult: () -> Unit = {},
+    onPermissionsLaunch: (Set<String>) -> Unit = {}
+) {
+    // Remember the last error ID, such that it is possible to avoid re-launching the error
+    // notification for the same error when the screen is recomposed, or configuration changes etc.
+    val errorId = rememberSaveable { mutableStateOf(UUID.randomUUID()) }
+
+    LaunchedEffect(uiState) {
+        // If the initial data load has not taken place, attempt to load the data.
+        if (uiState is RecordListScreenViewModel.UiState.Uninitialized) {
+            onPermissionsResult()
+        }
+
+        // The [RecordListScreenViewModel.UiState] provides details of whether the last action
+        // was a success or resulted in an error. Where an error occurred, for example in reading
+        // and writing to Health Connect, the user is notified, and where the error is one that can
+        // be recovered from, an attempt to do so is made.
+        if (uiState is RecordListScreenViewModel.UiState.Error && errorId.value != uiState.uuid) {
+            onError(uiState.exception)
+            errorId.value = uiState.uuid
+        }
+    }
+
+    if (uiState != RecordListScreenViewModel.UiState.Uninitialized) {
+        LazyColumn(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.Top,
+            /*horizontalAlignment = Alignment.CenterHorizontally*/ ) {
+            if (!permissionsGranted) {
+                item {
+                    Button(onClick = { onPermissionsLaunch(permissions) }) {
+                        Text(text = stringResource(R.string.permissions_button_label))
+                    }
+                }
+            } else {
+                item {
+                    Text(
+                        text = "${recordType.clazz.simpleName} [$uid]",
```

Comment:
Will this always be dynamic? Can the text be updated to use [stringResource](https://developer.android.com/develop/ui/compose/resources)?

Or do we think it will be more common for developers to not have fixed data types they read in?

Reply:
This will be somewhat dynamic since it depends on the class of the passed in recordType